src/GraphQL/Validation/Rules/ArgumentsOfCorrectType.cs

```diff
@@ -28,7 +28,7 @@ public class ArgumentsOfCorrectType : IValidationRule
                 return;
 
             var type = argDef.ResolvedType;
-            var errors = type.IsValidLiteralValue(argAst.Value ?? argDef.GetDefaultValueAST(context.Schema), context.Schema);
+            var errors = type.IsValidLiteralValue(argAst.Value, context.Schema);
```

Comment:
If `argAst.Value` is null, then it is specified to be null, and should not use the default value instead.  So the previous check was incorrect.

Reply:
👍  Need to write a test to verify.

Reply:
#2339

Reply:
ok